scripts/base/misc/version.zeek

```diff
@@ -92,6 +92,8 @@ function parse(version_string: string): VersionDescription
 
 		vs = gsub(vs, /-debug$/, "");
 		vs = gsub(vs, /-(beta|rc|dev)[0-9]*/, "");
+		# Drop the optional local version piece
+		vs = gsub(vs, /-[a-zA-Z0-9]+$/, "");
```

Comment:
How about we add a field `localversion` to the record and fill that in too?

Reply:
While working on this, I think I've found some abiguity. In the version string "2.5.2-beta5-12-debug", what does the "12" signify? And, how could that be distinguished from a localversion?

Reply:
It seems that "12" is the number of commits relative to the beta|rc|dev. And, it seems that a "." is currently used between the beta|rc|dev and the commit count. If that's all true, I'd propose disallowing a "-" as a separator for that grouping. Perhaps there should be a "rule" that a "-" is only used to separate the extra portions of the version string? (beta|dev|rc, localversion, debug). @ckreibich thoughts?

Reply:
Nice catch Peter. Yes, that rule is the way to go. We changed the dev versioning scheme in 2019, [here](https://github.com/zeek/zeek-aux/commit/6b9ab346f9243270e25d41bdd67c62b9cd5fc2d7), so "2.5.2-beta5-12-debug" would not occur today — it would be "2.5.2-beta5.12-debug". So I think proceeding as planned and enforcing that localversion doesn't contain dashes is all we need.

Reply:
Thanks Peter! That macOS failure is unrelated to your PR and we'll sort it out.

Reply:
That was fixed by https://github.com/zeek/zeek/pull/3676. This just needs to be rebased to pick it up.

Reply:
Yeah I just saw. Thank you both for being so quick. I'm approving.